### ISSUE TITLE
[DeviceClient] SetSASTokenExpiryTime doesn't work for first sas token

### ISSUE THREAD

**Issue by timtay-microsoft** [Azure/azure-iot-sdk-java#317]
- **OS and version used:** any

- **Java runtime used:** any

- **SDK version used:** any

# Description of the issue:

Setting the sas token expiry time before calling open does not set the sas token expiry time of the first generated sas token. Even though the first sas token is not needed until open is called, it is generated during the DeviceClient constructor anyways. The sas token should not be generated until open() is called so that a user can alter the expiry time of that first sas token.

# Code sample exhibiting the issue:
```java
    DeviceClient client = new DeviceClient(connString, protocol);
    client.setOption("SetSASTokenExpiryTime", 10L);
    client.open(); //connection won't expire until the default 60 minutes have passed
```

**Comment by timtay-microsoft** [comment]
Thanks for filing this issue. We'll look into it soon.

**Comment by timtay-microsoft** [comment]
The fix for this issue has been checked into master now, and is slated to be released on 8/30.

**Comment by timtay-microsoft** [comment]
The fix for this issue has been released as of Device Client 1.14.0, so I'm closing this issue now.
